playwright/_impl/_helper.py

```diff
@@ -142,27 +142,26 @@ class FrameNavigatedEvent(TypedDict):
 Env = Dict[str, Union[str, float, bool]]
 
 
-class URLMatcher:
-    def __init__(self, base_url: Union[str, None], match: URLMatch) -> None:
-        self._callback: Optional[Callable[[str], bool]] = None
-        self._regex_obj: Optional[Pattern[str]] = None
-        if isinstance(match, str):
-            if base_url and not match.startswith("*"):
-                match = urljoin(base_url, match)
-            regex = glob_to_regex(match)
-            self._regex_obj = re.compile(regex)
-        elif isinstance(match, Pattern):
-            self._regex_obj = match
-        else:
-            self._callback = match
-        self.match = match
-
-    def matches(self, url: str) -> bool:
-        if self._callback:
-            return self._callback(url)
-        if self._regex_obj:
-            return cast(bool, self._regex_obj.search(url))
-        return False
+def url_matches(
+    base_url: Optional[str], url_string: str, match: Optional[URLMatch]
+) -> bool:
+    if not match:
+        return True
+    if isinstance(match, str) and match[0] != "*":
+        # Allow http(s) baseURL to match ws(s) urls.
+        if (
+            base_url
+            and re.match(r"^https?://", base_url)
+            and re.match(r"^wss?://", url_string)
+        ):
+            base_url = re.sub(r"^http", "ws", base_url)
+        if base_url:
+            match = urljoin(base_url, match)
+    if isinstance(match, str):
+        match = glob_to_regex(match)
+    if isinstance(match, Pattern):
+        return bool(match.search(url_string))
+    return match(url_string)
```

Comment:
For some reason Python does not have this code:

https://github.com/microsoft/playwright/blob/f43b86fea43aeb99cfea4a0773c577dd9801b35d/packages/playwright-core/src/utils/isomorphic/urlMatch.ts#L110-L118

The "past path" seems an optimisation while the 'parseURL path matching' seems not implemented at all?

Reply:
It seems like js has some dead code. `match` cannot be a string there because we do `match = globToRegex(match);` before.